packages/react-components/src/echarts/extensions/minMaxDataStreamValues/updateDataStreamMinMax.ts

```diff
@@ -0,0 +1,20 @@
+import LineSeriesModel from 'echarts/types/src/chart/line/LineSeries';
+import { EChartsExtensionInstallRegisters } from 'echarts/types/src/extension';
+import { findMinMax, handleMinMaxSync } from './minMaxDataStreamSync';
+
+// Echarts core use type does not map correctly to the echarts extension type so exporting as any
+// eslint-disable-next-line
+export const dataStreamMinMaxSyncExtension: any = (registers: EChartsExtensionInstallRegisters) => {
+  registers.registerUpdateLifecycle(
+    'series:afterupdate',
+    (_ecModel, _api, params) => {
+      if (!params.updatedSeries) return;
+
+      params.updatedSeries.forEach((series) => {
```

Comment:
have you tested adding / removing datastreams? Sometimes echarts can have null series values.

Reply:
Yes! I have a case where it is a null value it should store undefined, and then display "No max found" or something in the legend or "-" to indicate its not in the viewport